config/jobs/kubernetes-sigs/vsphere-csi-driver/vsphere-csi-driver.yaml

```diff
@@ -92,6 +92,26 @@ presubmits:
       testgrid-alert-email: k8s-testing-cloud-provider-vsphere+alerts@groups.vmware.com
       description: Verifies the markdown has been linted
 
+  # Runs 'staticcheck' on appropriate sources
+  - name: pull-vsphere-csi-driver-verify-staticcheck
+    always_run: false
+    run_if_changed: '^((cmd|pkg)/)|hack/check-staticcheck\.sh'
+    optional: true
```

Comment:
I don't know a lot about your particular workflow, so I have a couple of questions about this

A job that is run "optionally" won't block a merge. I know you're trying to get some info on how this runs, but what confidence will you get that you wouldn't by running the job or `make` command locally?

Would it be easier to block CNS code that doesn't pass, instead of maybe going back and making it compliant afterward?

Reply:
All good questions. 

This is a temporary setting for us. There is some code coming into this particular repo, and it is coming in pieces (for.... reasons), and my concern is that the static checker will flag code as "un-used". And in truth it will be until all the code comes in. so I'm trying to prevent that, but get early feedback on issues that might arise.

Reply:
That makes sense. As long as the alert email and dashboard are being monitored, it seems unlikely this test will just be forgotten.

Is there even a rough estimate to how long the unpleasantness will last? If there is, it might be worth adding a tombstone comment here. ex `# Temporary for ...reasons. Remove this after Jan 1 2020`

Reply:
Good point... But, roughly two weeks. It's something I will be looking to remove, so it's not something I will forget. :)